### PR TITLE
go/services/regulated-assets-approval-server/internal/db: Build migrations into image.

### PR DESCRIPTION
**Pull request by Zagan202**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

- [x] Revert dockerfile change
- [x] Build the migrations into the binary, like [other services](https://github.com/stellar/go/tree/master/exp/services/recoverysigner/internal/db/dbmigrate) do. 

### Why

Fixes issue with migrations during deployment, where instead of copying sql files from a hardcoded path we instead build them into the image.  

### Known limitations

N/A

### CODE REVIEW

File: services/regulated-assets-approval-server/internal/db/dbmigrate/dbmigrate_test.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	require.NoError(t, err)
+
+	n, err := Migrate(session, migrate.Up, 2)
+	require.NoError(t, err)
+	require.Equal(t, 2, n)
+
+	migrations, err := PlanMigration(session, migrate.Down, 0)
+	require.NoError(t, err)
+	wantMigrations := []string{
+		"accounts-kyc-status.sql",
```
Comment (leighmcculloch) [inline services/regulated-assets-approval-server/internal/db/dbmigrate/dbmigrate_test.go:97]: This is unrelated to this PR specifically, but I think you'll need to rename this migration file. Migration files are, I think, read in the order that they sort by their filename, that's why the initial starts with a date and an order number. If you use migration files with only words the migrations won't be guaranteed to apply in the same order in every environment.

You might not be able to change a migration name after it has been applied in an environment if the migration system uses the name to know if it has been applied before or not.

**Review by leighmcculloch — APPROVED** [review]
Looks good, although I note one issue with the `accounts-kyc-status.sql` migration that is worth fixing before it gets applied to any environment, if possible.

File: services/regulated-assets-approval-server/README.md [hunk 2]
```diff
@@ -59,6 +60,14 @@ Flags:
       --database-url string   Database URL (DATABASE_URL) (default "postgres://localhost:5432/?sslmode=disable")
 ```
 
+#### Migration files
+
+regulated-assets-approval-server builds the migrations into the binary. If there are any changes to the db(adding, removing or updating tables) generate a new `internal/db/dbmigrate/dbmigrate_generated.go` using the [bindata](https://github.com/jteeuwen/go-bindata) package.
+
+```sh
+$ go-bindata -o dbmigrate_generated.go internal/db/dbmigrate/migrations/
```
Comment (leighmcculloch) [inline services/regulated-assets-approval-server/README.md:68]: Running `./gogenerate.sh` in the root of the repo should also do it as long as you've setup the `//go:generate` comment.

**Review by leighmcculloch — CHANGES_REQUESTED** [review]
Actually, which go-bindata did you generate the file with? I note in teh README you reference a tool that's different to the one we're using.

We use this one:
https://github.com/stellar/go/blob/4e3c550a5492cab844f14c71fabce1f3d5c8ffa8/gogenerate.sh#L9

Unfortunately there are multiple tools with the same name.

File: services/regulated-assets-approval-server/internal/db/dbmigrate/dbmigrate_test.go (hunk 1 above)
Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/db/dbmigrate/dbmigrate_test.go:97]: I'll change this in a subsequent PR.

Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/db/dbmigrate/dbmigrate_test.go]: Changed name here 5e6315c

File: services/regulated-assets-approval-server/internal/db/dbtest/dbtest.go [hunk 3]
```diff
@@ -9,13 +9,18 @@ import (
...
 
 	// Get the folder holding the migrations relative to this file. We cannot
-	// hardcode "../dbmigrate/migrations" because Open is called from tests in
-	// multiple packages and tests are executed with the current working
-	// directory set to the package the test lives in.
+	// hardcode "../migrations" because Open is called from tests in multiple
+	// packages and tests are executed with the current working directory set
+	// to the package the test lives in.
 	_, filename, _, _ := runtime.Caller(0)
 	migrationsDir := path.Join(path.Dir(filename), "..", "dbmigrate", "migrations")
```
Comment (marcelosalloum) [inline services/regulated-assets-approval-server/internal/db/dbtest/dbtest.go:25]: 🤔  I'm a little confused here: you updated the path in the comment from "../dbmigrate/migrations" to "../migrations" but the migrations directory and the migration sql files are still in "../dbmigrate/migrations". Was this comment wrongly updated?

Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/db/dbtest/dbtest.go:25]: Yes this comment was wrongly updated, standby.

Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/db/dbtest/dbtest.go]: Fixed here b499399

File: services/regulated-assets-approval-server/README.md [hunk 4]
```diff
@@ -59,6 +60,14 @@ Flags:
       --database-url string   Database URL (DATABASE_URL) (default "postgres://localhost:5432/?sslmode=disable")
 ```
 
+#### Migration files
+
+regulated-assets-approval-server builds the migrations into the binary. If there are any changes to the db(adding, removing or updating tables) generate a new `internal/db/dbmigrate/dbmigrate_generated.go` using the `gogenerate.sh` script located at the root of the repo.
+
+```sh
+$ ./gogenerate.sh internal/db/dbmigrate/migrations/
```
Comment (leighmcculloch) [inline services/regulated-assets-approval-server/README.md:68]: The script doesn't use any arguments passed to it.
```suggestion
$ ./gogenerate.sh
```

**Review by leighmcculloch — APPROVED** [review]
One comment, otherwise looks great.

**Pull request merged**